### ISSUE TITLE
Time Travel Debugging

### ISSUE THREAD

**Issue by rugk** [hbenl/vscode-firefox-debug#87]
Firefox Nightly supports time travel debugging, see https://wiki.mozilla.org/Time_Travel_Debugging and https://www.bleepingcomputer.com/news/software/firefox-is-testing-time-travel-debugging/.

So you may want to do so, too. :smile:

**Comment by hbenl** [comment]
That sounds awesome. I hope they make it available on other platforms soon, because I don't have a Mac...

**Comment by rugk** [comment]
AFAIK it's not limited to Mac, just to FF Nightly. Bleepingcomputers just wrong stupid stuff… :laughing: 

https://gist.github.com/jasonLaster/1e220992c294a571dd9b59abd084ccf2#getting-started

At least i tried it on Linux and I had to create the about:config value, but the recording button was already there. Could not find out how to actually travel back then thopugh.

**Comment by rugk** [comment]
Okay, wrong: As stated [here](https://gist.github.com/jasonLaster/1e220992c294a571dd9b59abd084ccf2#gistcomment-2663168) only the menu is added on Linux, the replay itself is really currently limited to Mac OS… well… I hope we'll get it soon. :smi

**Comment by crazyquark** [comment]
hello. was this implemented?

**Comment by hbenl** [comment]
@crazyquark No, this is only available in Firefox for Mac and I haven't seen any news about this for a year...
If this becomes a standard feature on all platforms, I will add support for it to this adapter.

**Comment by jasonLaster** [comment]
Hi, here is the current [roadmap](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/WebReplayRoadmap) it is currently experimental and not planned for release any time soon.

**Comment by jasonLaster** [comment]
It would be great to add support for replay in the adapter.

The way I see this working, in vscode's config you could enable replay, which would open firefox with recording available. When replay is enabled, vs code would add rewind and step back.

Here is a replay bug for adding any firefox functionality we need
https://bugzilla.mozilla.org/show_bug.cgi?id=1581948

NOTE: I believe vs code has support for rewind and step back.

**Comment by jasonLaster** [comment]
Here is a useful link:

https://github.com/microsoft/vscode-debugadapter-node/blob/master/protocol/src/debugProtocol.ts#L747-L754

**Comment by jasonLaster** [comment]
supportsStepback will do this

https://github.com/microsoft/vscode-debugadapter-node/blob/master/protocol/src/debugProtocol.ts#L1353

https://twitter.com/jasonlaster11/status/1174091521345257473

**Comment by jasonwilliams** [comment]
Hey @jasonLaster what happened with this?

**Comment by jasonLaster** [comment]
We ended up taking web replay out of firefox.

Brian and I are working on something similar outside of mozilla though.
